WordPress/Classes/ViewRelated/Site Creation/Final Assembly/SiteAssemblyWizardContent.swift

```diff
@@ -135,7 +138,31 @@ final class SiteAssemblyWizardContent: UIViewController {
                 // making ANY modification to this stat please refer to: p4qSXL-35X-p2
                 SiteCreationAnalyticsHelper.trackSiteCreationSuccess(self.siteCreator.design)
             }
-            self.contentView.status = status
+            if status == .succeeded,
+               shouldPerformDomainPurchasingStep,
+               let domain = siteCreator.address,
+               let blog = self.createdBlog {
+                self.attemptDomainPurchasing(domain: domain, site: blog)
+            } else {
+                self.contentView.status = status
+            }
+        }
+    }
+
+    /// The site must be created before attempting domain purchasing.
+    private func attemptDomainPurchasing(domain: DomainSuggestion, site: Blog) {
+        self.domainPurchasingController.purchase(domain: domain, site: site) { [weak self] result in
+            guard let self else {
+                return
+            }
+            switch result {
+            case .success(let domain):
+                self.contentView.siteName = domain
+            case .failure:
+                // TODO: We should discuss how to handle domain purchasing errors
```

Comment:
I suppose we don't have designs for this?

Reply:
Not yet, but it will be handled later. See internal discussion: p1681907249388189-slack-C04M1NX8YCR